CMakeLists.txt

```diff
@@ -0,0 +1,200 @@
+# - Main application (M/Mahogany executable)
+# - IMAP library for mail handling
+# - Compface library for X-Face image support
+# - DSPAM library for spam filtering
+# - Versit library for vCard handling
+# - Python scripting interface
+#
+# USAGE:
+# ------
+# 1. Prerequisites:
+#    - CMake 3.16 or later
+#    - C++ compiler with C++11 support
+#    - wxWidgets 3.0 or later
+#    - Python development headers (optional, for Python support)
+#    - OpenSSL (optional, for SSL/TLS support)
+#
+# 2. Basic build:
+#    mkdir build && cd build
+#    cmake ..
+#    make
+#
+# 3. Build with options:
+#    cmake -DBUILD_SHARED_LIBS=ON -DENABLE_PYTHON=ON -DCMAKE_BUILD_TYPE=Release ..
+#    make
+#
+# 4. Install:
+#    make install
+#
+# BUILD OPTIONS:
+# --------------
+# - BUILD_SHARED_LIBS: Build shared libraries instead of static (default: OFF)
+# - ENABLE_PYTHON: Enable Python scripting support (default: ON if Python found)
+# - ENABLE_SSL: Enable SSL/TLS support (default: ON if OpenSSL found)
+# - CMAKE_BUILD_TYPE: Debug, Release, RelWithDebInfo, MinSizeRel
+# - CMAKE_INSTALL_PREFIX: Installation directory
+#
+# TARGETS:
+# --------
+# Libraries:
+# - mahogany_imap: IMAP client library (static/shared)
+# - mahogany_compface: X-Face image library (static/shared)
+# - mahogany_dspam: Spam filtering library (static/shared)
+# - mahogany_versit: vCard library (static/shared)
+#
+# Executables:
+# - mahogany: Main GUI application
+# - mtest: IMAP test utility (if built)
+# - mailutil: Mail utility (if built)
+#
+# PLATFORM SUPPORT:
+# ------------------
+# - Windows (MSVC, MinGW)
+# - Linux (GCC, Clang)
+# - macOS (Clang, GCC)
+# - Other Unix-like systems
+#
+# The CMake configuration automatically detects the platform and applies
+# appropriate compiler flags and library dependencies.
+#
+#################################################################################
+
+cmake_minimum_required(VERSION 3.16)
+
+project(Mahogany 
+    VERSION 0.68.0
+    DESCRIPTION "Mahogany Cross-Platform Email Client"
+    LANGUAGES C CXX
+)
+
+# Set C++ standard
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+# Include CMake modules
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+# Build options
+option(BUILD_SHARED_LIBS "Build shared libraries" OFF)
+option(ENABLE_PYTHON "Enable Python scripting support" ON)
+option(ENABLE_SSL "Enable SSL/TLS support" ON)
+option(BUILD_UTILITIES "Build utility programs" ON)
+
+# Set default build type if not specified
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING 
+        "Choose the type of build: Debug Release RelWithDebInfo MinSizeRel" FORCE)
+endif()
+
+# Platform-specific settings
+if(WIN32)
+    # Windows-specific settings
+    add_definitions(-DWIN32 -D_WINDOWS -D_UNICODE -DUNICODE)
+    if(BUILD_SHARED_LIBS)
+        add_definitions(-DWXUSINGDLL)
+    endif()
```

Comment:
This is incorrect and should be removed, `WXUSINGDLL` comes from wxWidgets detection.